lte/gateway/c/core/oai/tasks/s1ap/s1ap_mme_handlers.c

```diff
@@ -1215,20 +1223,33 @@ status_code_e s1ap_mme_handle_ue_context_release_request(
         (uint32_t) mme_ue_s1ap_id, (uint32_t) enb_ue_s1ap_id);
     OAILOG_FUNC_RETURN(LOG_S1AP, RETURNerror);
   } else {
-    if (ue_ref_p->enb_ue_s1ap_id == enb_ue_s1ap_id) {
+    s1ap_imsi_map_t* imsi_map = get_s1ap_imsi_map();
+    hashtable_uint64_ts_get(
+        imsi_map->mme_ue_id_imsi_htbl, (const hash_key_t) mme_ue_s1ap_id,
+        &imsi64);
+    if (ue_ref_p->sctp_assoc_id == assoc_id &&
+        ue_ref_p->enb_ue_s1ap_id == enb_ue_s1ap_id) {
```

Comment:
as a nit, I find it more tractable and less error prone to have each predicate within their own parenthesis.